templates/kotlin/src/main/kotlin/io/appwrite/models/Payload.kt.twig

```diff
@@ -0,0 +1,67 @@
+package io.appwrite.models
+
+import io.appwrite.extensions.gson
+import java.io.File
+import java.net.URLConnection
+import java.nio.file.Files
+import java.nio.file.Paths
+
+class Payload private constructor() {
+
+    lateinit var path: String
+    lateinit var filename: String
+    lateinit var mimeType: String
+    lateinit var sourceType: String
+    lateinit var data: Any
+
+    override fun toString(): String {
+        if (sourceType != "bytes") {
+            throw IllegalArgumentException("source type is not supported: $sourceType")
+        }
+
+        return String(data as ByteArray)
+    }
+
+    fun toBinary(): ByteArray {
+        if (sourceType != "bytes") {
+            throw IllegalArgumentException("source type is not supported: $sourceType")
+        }
+
+        return data as ByteArray
+    }
+
+    fun toJson(): MutableMap<String, Any> {
+        if (sourceType != "bytes") {
+            throw IllegalArgumentException("source type is not supported: $sourceType")
+        }
+
+        return gson.fromJson(toString(), MutableMap::class.java) as MutableMap<String, Any>
+    }
+
+
+    companion object {
+        fun fromFile(file: File) = Payload().apply {
+            path = file.canonicalPath
+            filename = file.name
+            mimeType = Files.probeContentType(Paths.get(file.canonicalPath))
+                ?: URLConnection.guessContentTypeFromName(filename)
+                        ?: ""
+            sourceType = "file"
+        }
+
+        fun fromPath(path: String): Payload = fromFile(File(path)).apply {
+            sourceType = "path"
+        }
+
+        fun fromBinary(bytes: ByteArray, filename: String = "", mimeType: String = "") = Payload().apply {
```

Comment:
mimeType doesnt follow structure we decided. Let me know if it's necessary for implementation, i which case we might need to apply it to all SDKs. otherwise, lets follow this, and update all methods accordingly:

```
function toBinary(): bytes[]
function toJson(): JSON
function toString(): string
function toFile(path: string): void

static function fromBinary(bytes: bytes[], name?: string): Payload
static function fromFile(file: File, name?: string): Payload // Name truly optional in Storage
static function fromJson(object: JSON, name?: string): Payload
static function fromString(text: string, name?: string): Payload
```